backend/server/lib/utils.js

```diff
@@ -567,9 +568,37 @@ function objectMemSize(object) {
     return object.toString().length*2;
 }
 
+/**
+ * Zips the contents of a folder into a .zip file.
+ * 
+ * @param {string} sourceFolderPath - The path of the folder to zip.
+ * @param {string} zipFilePath - The path where the zip file should be saved.
+ * @returns {Promise<void>} - Resolves when the folder is successfully zipped.
+ * @throws {Error} - Throws an error if zipping fails.
+ */
+
+async function zipFolder(sourceFolderPath, zipFilePath) {
+    const archive = archiver('zip', { zlib: { level: 9 } });
+
+    archive.directory(sourceFolderPath, false);
+
+    return new Promise((resolve, reject) => {
+        archive.on('error', reject);
```

Comment:
I never said don't log error if you are rejecting then why aren't you rejecting with the error thrown?